wp/sections/work.tex

```diff
@@ -0,0 +1,129 @@
+                    child {node {LPAR}}
+                    child {node {VALUE\underline{ }ARGUMENT}
+                        child {node {STRING\underline{ }TEMPLATE}
+                            child {node {OPEN\underline{ }QUOTE}}
+                            child {node {LITERAL\underline{ }STRING\underline{ }TEMPLATE\underline{ }ENTRY}
+                                child {node {REGULAR\underline{ }STRING\underline{ }PART}}
+                            }
+                            child [missing] {}
+                            child {node {CLOSING\underline{ }QUOTE}}
+                        }
+                    }
+                    child [missing] {}
+                    child [missing] {}
+                    child [missing] {}
+                    child [missing] {}
+                    child [missing] {}
+                    child [missing] {}
+                    child {node {RPAR}}
+                }
+            }
+            child [missing] {}
+            child [missing] {}
+            child [missing] {}
+            child [missing] {}
+            child [missing] {}
+            child [missing] {}
+            child [missing] {}
+            child [missing] {}
+            child [missing] {}
+            child [missing] {}
+            child [missing] {}
+            child [missing] {}
+            child [missing] {}
+            child {node {WHITE\underline{ }SPACE}}
+            child {node {RBRACE}}
+        }
+    };
+\end{tikzpicture}
+
+If there are error elements inside the constructed tree, then the corresponding error is displayed. Further, for each rule in the ruleset, the \textsl{visit} method is called to which the root node itself and its “children” are sequentially passed.
+At startup, you can pass flags to ktlint - one of them is "-F". This flag means that the rule will not only report an error, but try to fix it.
+
+\subsection{diKTat}
+\par
+Another feature of ktlint is that at startup you can transfer a JAR file with additional ruleset(s), and then all nodes will be passed to these rules. This is diKTat! DiKTat is a set of easily configurable rules for static code analysis. 
+The set of all rules is described in the \textsl{DiktatRuleSetProvider}\footnote{\url{https://github.com/cqfn/diKTat/blob/master/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/DiktatRuleSetProvider.kt}} class. This class overrides the \textsl{get()} method of the \textsl{RuleSetProvider}\footnote{\url{https://github.com/pinterest/ktlint/blob/master/ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/RuleSetProvider.kt}} interface, which returns a set of rules to be "traversed". But before returning this set, the configuration file, in which the user has independently configured all the rules, is read. If there is no configuration file, then a warning will be displayed and the rules will be triggered in according with the default configuration file. 
```

Comment:
I think we should specify tags in github links instead of just pointing them to master. What if we move this class in the next release, and link becomes invalid?